main/webapp/modules/core/scripts/project/browsing-engine.js

```diff
@@ -240,19 +241,31 @@ BrowsingEngine.prototype.update = function(onDone) {
 
   this._elmts.header.show();
   this._elmts.controls.css("visibility", "hidden");
-  this._elmts.indicator.css("visibility", "visible");
+  this._elmts.indicator.css("display", "block");
 
   $.post(
     "command/core/compute-facets?" + $.param({ project: theProject.id }),
     { engine: JSON.stringify(this.getJSON(true)) },
     function(data) {
+      if(data.code === "error") {
+        var clearErr = $('#err-text').remove();
+        var err = $('<div id="err-text">')
+                  .text(data.message)
+                  .appendTo(self._elmts.errors);
+         self._elmts.errors.css("display", "block");
+        if (onDone) {
+          onDone();
+        }
+        return;
+      }
```

Comment:
In principle I'd prefer an `if () { ... } else { ... }` rather than `if () { ... ; return; } ...` but it's clearly not a big deal…